### PR TITLE
Lightning docker image based on base-cuda

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/pull/3074#pullrequestreview-486676042

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

cc: @Borda

### CODE REVIEW

File: .github/workflows/ci_dockers.yml [hunk 1]
```diff
@@ -14,8 +14,11 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python_version: [3.7, 3.8]
-        pytorch_version: [1.5]
+        python_version: [3.6, 3.7, 3.8]
```
Comment (Borda) [inline .github/workflows/ci_dockers.yml:17]: not sure if weed to test all but with base-cuda it shall be fast and without any extra installation, right?

**Review by Borda — APPROVED** [review]
love this chaining :]

File: .github/workflows/ci_dockers.yml [hunk 2]
```diff
@@ -14,8 +14,11 @@ jobs:
...
+        pytorch_version: [1.3, 1.4, 1.5, 1.6, 1.7]
```
Comment (Borda) [inline .github/workflows/ci_dockers.yml:18]:
```suggestion
        pytorch_version: [1.3, 1.5, 1.7]
```

File: .github/workflows/ci_dockers.yml (hunk 1 above)
Comment (Borda) [inline .github/workflows/ci_dockers.yml:17]:
```suggestion
        python_version: [3.6, 3.8]
```

Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml:17]: Yes, no extra installation

File: .github/workflows/ci_dockers.yml (hunk 2 above)
Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml]: Not testing all supported versions?

File: .github/workflows/ci_dockers.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci_dockers.yml]: but still it occupy workers which we have limited so lets lower a bit the load...
mind check if we can cache docker images with GH action? could be in another PR...

Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml]: Okay, will check it.

File: .github/workflows/ci_dockers.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci_dockers.yml]: we are testing it with each in regular CI, this is doing just installation, so no need for so dense testing just edge-cases

Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml]: Alright.

File: .github/workflows/ci_dockers.yml (hunk 1 above)
Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml]: Shall we cache base-cuda or conda? or both?

Comment (Borda) [reply .github/workflows/ci_dockers.yml]: ideally both, but let's move it to separate PR and merge this as it is

Comment (ydcjeff) [reply .github/workflows/ci_dockers.yml]: Yes, it will be in a separate PR, this is just asking in advance.

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]
LGTM
